lib/eventide/rails/database_tasks.rb

```diff
@@ -36,35 +61,39 @@ def wrap_ar_config
         ActiveRecord::Base.configurations = config
       end
 
-      def migrate(config)
-        begin
-          connection = ActiveRecord::Base.establish_connection(config).connection
-        rescue ActiveRecord::NoDatabaseError
+      def init_database(config)
+        connection = connection_for(config) do
           $stderr.puts "Eventide cannot be initialized for #{config['database']}: database not created"
           return
         end
-        if migrated?(connection)
+
+        if ready?(connection)
           puts "Eventide for '#{config['database']}' already initialized"
           return
         end
+
+        if messages_table_present?(connection)
+          puts "#{config['database']}: cannot initialize, table messages already exists"
+          puts "Did you mean `rake es:update`?"
+        end
+
         execute_script script_path('extensions.sql'), connection
+        install_functions(connection)
         execute_directory('table', connection)
-        execute_directory('types', connection)
-        %w[
-          hash-64
-          category
-          stream-version
-          write-message
-          get-stream-messages
-          get-category-messages
-          get-last-message
-        ].each do |fn|
-          execute_script script_path("functions/#{fn}.sql"), connection
-        end
         execute_directory('indexes', connection)
         execute_directory('views', connection)
       end
 
+      def update_database(config)
+        connection = connection_for(config) do
+          $stderr.puts "Eventide cannot be updated for #{config['database']}: database not created"
+          return
+        end
+
+        install_functions(connection)
```

Comment:
After I've had some discussions with Scott, he's since launch another version of the postgres gem where it executes these three scripts in order to update:

```
# Install functions
source $base/install-functions.sh

# Install indexes
source $base/install-indexes.sh

# Install views
source $base/install-views.sh
```